ni/protobuf/types/parametric_data.proto

```diff
@@ -0,0 +1,62 @@
+//---------------------------------------------------------------------
+//---------------------------------------------------------------------
+syntax = "proto3";
+
+//---------------------------------------------------------------------
+//---------------------------------------------------------------------
+package ni.protobuf.types;
+
+//---------------------------------------------------------------------
+//---------------------------------------------------------------------
+option csharp_namespace = "NationalInstruments.Protobuf.Types";
+option go_package = "types";
+option java_multiple_files = true;
+option java_outer_classname = "ParametricDataProto";
+option java_package = "com.ni.protobuf.types";
+option objc_class_prefix = "NIPT";
+option php_namespace = "NI\\PROTOBUF\\TYPES";
+option ruby_package = "NI::Protobuf::Types";
+
+message ParametricDataValue {
+  oneof value {
+    double double_value = 1;
+    bool bool_value = 2;
+    string string_value = 3;
+    int32 int_value = 4;
+  }
+  string label = 5;
+  string unit = 6;
+}
+
+message ParametricDataValueArray {
+    message DoubleArray {
```

Comment:
Something else to consider is whether we should extract these nested message definitions into `array.proto` and document them as convenience messages for use in oneof fields or declaration of 2D jagged arrays.